packages/headless/src/composables/typeAhead.ts

```diff
@@ -0,0 +1,26 @@
+import { isPrintableCharacter } from "../utils/keyEvent";
+import { debounce } from "../utils/timer";
+
+/**
+ * Enhances typeahead to combine multiple inputs in quick succession and filter out non-printable characters.
+ *
+ * @example
+ * ```ts
+ * const typeAhead = useTypeAhead((inputString) => onTypeAhead(inputString));
+ * // ...
+ * const onKeydown = (key: KeyboardEvent) => typeAhead(key.event);
+ * ```
+ */
+export const useTypeAhead = (callback: (input: string) => void) => {
+  let inputString = "";
+  const debouncedReset = debounce(() => (inputString = ""), 500);
+
+  return (event: KeyboardEvent) => {
+    if (!isPrintableCharacter(event.key)) {
+      return;
+    }
+    debouncedReset();
```

Comment:
each letter will trigger a new timeout, is that correct? should we cancel a previous timeout if there is one before starting a new one?

Reply:
we might need a longer word in the listbox to test this

Reply:
I changed it to `vi.runAllTimers();` so we do not rely on timeouts